idxmgmt/ilm/policy.go

```diff
@@ -0,0 +1,97 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package ilm
+
+var eventPolicies = map[string]map[string]interface{}{
```

Comment:
I wasn't sure about using a map here since it's static so tried it out with a struct - it's not any better IMO.  details below in case you want to consider it still.

<details>

```go
type policy map[string]interface{}

type EventPolicies struct {
	error       policy
	span        policy
	transaction policy
	metric      policy
}

var eventPolicies = EventPolicies{
	error:       PolicyKeepMedium,
	span:        PolicyKeepMedium,
	transaction: PolicyKeep,
	metric:      PolicyKeep,
}

func (e *EventPolicies) get(eventType string) (policy, error) {
	switch eventType {
	case "error":
		return e.error, nil
	case "span":
		return e.span, nil
	case "transaction":
		return e.transaction, nil
	case "metric":
		return e.metric, nil
	default:
		return nil, fmt.Errorf("unknown event type: %s", eventType)
	}
}
```

</details>